### ISSUE TITLE
emTripLog unexpected behavior on Android phones when phone dies or is turned off

### ISSUE THREAD

**Issue by manuelsantanap** [e-mission/e-mission-base#3]
When testing emTripLog on two low-end Samsung phones, I found an unexpected behavior when the phone either died or was turned off. Because the app only works with a link to a specific project, only those with such link can replicate the error described below. The link I used is of a project my colleague [Megumi Yamanaka](https://ced.berkeley.edu/academics/city-regional-planning/people/phd-students/megumi-yamanaka) from at U.C. Berkeley is working on).

In both cases, when I turned the phone back on, emTripLog did not continue tracking any activity. I reached that conclusion because neither the JSON file downloaded from the 'profile' tab nor the 'trips' tab showed any of the activities conducted after the phone died or was turned off. I checked the phone was connected to a secure and stable wifi network so the app could sync with the server for proper data transmission after reaching that conclusion.

Some unsuccessful actions to troubleshoot: 

1. First, I closed and opened the app, but nothing happened. 
2. Then, I turned the phone off and then on, but nothing happened. 
3. I tried to switch accounts, the app automatically went back to the previous account used (see video here: [emtriplog_issue2.zip](https://github.com/e-mission/e-mission-base/files/2221396/emtriplog_issue2.zip)). 

The app recorded data again only when I re-installed the application.  Unfortunately, I don't have access to the server, so I cannot confirm that the data was lost.

**Comment by shankari** [comment]
@manuelsantanap is this reproducible? Can you email me the logs (Profile -> Email Log)?
You can also attach them here, but that is essentially publishing them.
Since the logs contain privacy-sensitive location information, I'd suggest emailing them instead

**Comment by shankari** [comment]
@manuelsantanap thank you for sending me the logs. Can you also add a comment with a rough timeline of when you rebooted the phone, and when you took the trips that were not recorded?

**Comment by manuelsantanap** [comment]
Here is how I reproduced bug - The app stops tracking when the battery dies

Saturday, July 14 
- The battery of my Samsung Mini Prime died at 12:22 pm on my way home.
- The phone was charged that night.

Sunday, July, 15
- The app 'trips' tab showed no activities after the phone died - wifi and GPS were on 

Monday, July, 16
- I downloaded the 07/15 data (JSON file from phone - profile tab) in the morning (did not write up the time) and found no data associated with my activities   
- emTripLog + UI in Spanish was reinstalled at 5 PM
- The trips tap visualized trips after 5 PM this day

Bug reproduction
Monday, July, 16
- I let the battery of my second Samsung Mini Prime die by 5:30 pm
- Phone was charged some minutes later and forced syncing
- Downloaded JSON file from 07/16 that night. It has no data from 5:32:55 PM

**Comment by shankari** [comment]
For the record, @manuelsantanap emailed me new logs.
He reported that he "turned the phone off yesterday at 17:30 and turned it on couple of hours after".

**Comment by shankari** [comment]
Looking more closely:

Here's where we rebooted
```
4748,1532741718.826,2018-07-27 18:35:18.826000-07:53,BootReceiver : BootReceiver.onReceive called
```

We immediately launched a sync

```
4754,1532741754.883,2018-07-27 18:35:54.883000-07:53,ServerSyncAdapter : Starting sync with push
4761,1532741755.181,2018-07-27 18:35:55.181000-07:53,"ServerSyncAdapter : Push complete, now pulling"
```

And then sent out the initialize to start tracking again

```
4766,1532741755.333,2018-07-27 18:35:55.333000-07:53,"TripDiaryStateMachineRcvr : TripDiaryStateMachineReciever onReceive(android.app.ReceiverRestrictedContext@2c972ea, Intent { act=local.transition.initialize flg=0x10 cmp=edu.berkeley.eecs.embase/edu.berkeley.eecs.emission.cordova.tracker.location.TripDiaryStateMachineReceiver bqHint=4 }) called"
```

But the `TripDiaryStateMachineRcvr` didn't actually handle the initialize. The
next log message was an hour later, when we synced again.

```
4767,1532745306.8170002,2018-07-27 19:35:06.817000-07:53,BuiltinUserCache : Added value for key stats/client_nav_event at time 1.532745306806E9
4771,1532745306.9160001,2018-07-27 19:35:06.916000-07:53,ServerSyncAdapter : Starting sync with push
```

**Comment by shankari** [comment]
Looking through the code for `onReceive`, when we get an initialize transition, we check for consent and if successful, we print 

```
Log.i(context, TAG, reqConsent + " is the current consented version, sending msg to service...");
```

I don't see that message in the logs, but I don't see of the failure cases as well.
It looks like there was a crash while checking the consent.

This is probably because the consent checking caused this to crash. We first
read the consent from the `config.xml` file. As we know, that is `null`.

```
String reqConsent = getReqConsent(context);
```

And then in `ConfigManager.isConsented`, we do 

```
return currConfig != null && reqConsent.equals(currConfig.getApproval_date());
```

which is probably triggering an NPE.

**Comment by shankari** [comment]
Reproduced by force-generating `initialize` from the developer zone, which triggered this:

```
07-28 22:21:29.939  7796  7796 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
07-28 22:21:29.939  7796  7796 E AndroidRuntime: 	at edu.berkeley.eecs.emission.cordova.tracker.ConfigManager.isConsented(ConfigManager.java:60)
07-28 22:21:29.939  7796  7796 E AndroidRuntime: 	at edu.berkeley.eecs.emission.cordova.tracker.location.TripDiaryStateMachineReceiver.onReceive(TripDiaryStateMachineReceiver.java:100)
07-28 22:21:29.939  7796  7796 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3252)
07-28 22:21:29.939  7796  7796 E AndroidRuntime: 	... 8 more
```

**Comment by shankari** [comment]
Modified `config.xml` to include 

```
    <preference name="emSensorDataCollectionProtocolApprovalDate" value="2016-07-14" />
```

resent the `initialize`. Worked this time.

**Comment by shankari** [comment]
@manuelsantanap 
[em-base-issue-3-fix.apk.zip](https://github.com/e-mission/e-mission-base/files/2238397/em-base-issue-3-fix.apk.zip)

can you try whether this apk works? If so, I will generate a release version and upload to the stores. I am tempted to fix the prompt for the token-based authentication at the same time so that I don't have to keep going back for app store review.

**Comment by shankari** [comment]
From https://github.com/e-mission/e-mission-base/issues/3#issuecomment-408643870
> I am tempted to fix the prompt for the token-based authentication at the same time so that I don't have to keep going back for app store review.

I ended up fixing this as well. The `dummy-dev` mode is now called `prompted-auth` and you can specify the prompt to be used. Everything should still default to the dummy-dev prompt.

Here's the new apk for testing with the `prompted-auth` authentication mode and a custom prompt.

[em-base-issue-3-token-prompt-fix.apk.zip](https://github.com/e-mission/e-mission-base/files/2238832/em-base-issue-3-token-prompt-fix.apk.zip)

Unfortunately, the `megumibrt` channel connects to the main e-mission server which does not currently support token based auth. So you would really want to test this against your own channel that connects to your own token-based server.

Let me know when you are done testing, and I will push to the stores.

@manuelsantanap do you have an iPhone to test out the iOS changes, or can you ask Megumi to do it? Whoever does the change has to sign up for TestFlight - iOS does not allow installing arbitrary apks on the phone.

**Comment by manuelsantanap** [comment]
@shankari, I can do the tests also on my iPhone. Reporting to you back by tomorrow

**Comment by brianmickel** [comment]
Uninstalled App Store emTripLog from android phones.

We started with em-base-issue-3-token-prompt-fix.apk
- Hosted it at https://github.com/brianmickel/em-installs/blob/master/em-base-issue-3-token-prompt-fix.apk?raw=true
- Nagivated to the above link and downloaded apk to samsung android my files
- clicked through install screens
- resulted in "App Not Installed"

Repeated for em-base-issue-3-fix.apk
https://github.com/brianmickel/em-installs/blob/master/em-base-issue-3-fix.apk?raw=true
- resulted in "App Not Installed"

Repeated same steps for an apk of a different app on the android phone and installation was successful.

Is there a place the phone records logs during attempted installs we could send?

**Comment by shankari** [comment]
`adb logcat` is your friend.
Note that the `apk`s uploaded here are `.apk.zip` because github doesn't let me upload `.apk` files.
So you need to unzip them first.
If that still doesn't work, I will email you the apk files directly and let's see if that works better.

**Comment by manuelsantanap** [comment]
Both were unzipped. Here is the repo [link](https://github.com/brianmickel/em-installs) where Brian saved them. I appreciate if you could send me the apk files via email

**Comment by shankari** [comment]
@brianmickel thanks for letting me know. I think this is because of https://stackoverflow.com/a/46661987 - I just shared the apk that I used for testing my changes. I will make sure to rebuild before sharing apks in the future.

**Comment by shankari** [comment]
@manuelsantanap reported that he wasn't able to see trips after the update. From his logs:

There were initially no issues with calls to the server

```
103,1533064957.04,2018-07-31 12:22:37.040000-07:53,"ConnectionSettings : in getConnectURL, returning https://e-mission.eecs.berkeley.edu"
105,1533064958.175,2018-07-31 12:22:38.175000-07:53,CommunicationHelper : Got response org.apache.http.message.BasicHttpResponse@edf1967 with status HTTP/1.1 200 OK
```

```
108,1533064958.226,2018-07-31 12:22:38.226000-07:53,"ConnectionSettings : in getConnectURL, returning https://e-mission.eecs.berkeley.edu"
127,1533064958.842,2018-07-31 12:22:38.842000-07:53,CommunicationHelper : Got response org.apache.http.message.BasicHttpResponse@d9846fe with status HTTP/1.1 200 OK
```

```
131,1533064958.9420002,2018-07-31 12:22:38.942000-07:53,"ConnectionSettings : in getConnectURL, returning https://e-mission.eecs.berkeley.edu"
138,1533064959.5210001,2018-07-31 12:22:39.521000-07:53,CommunicationHelper : Got response org.apache.http.message.BasicHttpResponse@5bbfb62 with status HTTP/1.1 200 OK
139,1533064959.5279999,2018-07-31 12:22:39.528000-07:53,"CommunicationHelper : Result Summary JSON = {""timeline"": []}
 length 17"
```

But right around the time we read the timeline, we started getting proxy errors.

```
133,1533064958.95,2018-07-31 12:22:38.950000-07:53,"ConnectionSettings : in getConnectURL, returning https://e-mission.eecs.berkeley.edu"
142,1533064959.593,2018-07-31 12:22:39.593000-07:53,CommunicationHelper : Got response org.apache.http.message.BasicHttpResponse@b3325b0 with status HTTP/1.1 502 Proxy Error
```


```
176,1533064961.78,2018-07-31 12:22:41.780000-07:53,js : about to query for unprocessed trips from Tue Jul 31 2018 00:00:00 GMT-0700 -> Tue Jul 31 2018 23:59:59 GMT-0700
178,1533064961.795,2018-07-31 12:22:41.795000-07:53,"ConnectionSettings : in getConnectURL, returning https://e-mission.eecs.berkeley.edu"
183,1533064962.3879998,2018-07-31 12:22:42.388000-07:53,CommunicationHelper : Got response org.apache.http.message.BasicHttpResponse@9552f6b with status HTTP/1.1 502 Proxy Error
224,1533064962.965,2018-07-31 12:22:42.965000-07:53,js : About to dedup localResult = 0remoteResult = 0
225,1533064962.968,2018-07-31 12:22:42.968000-07:53,js : Deduped list = 0
226,1533064962.9720001,2018-07-31 12:22:42.972000-07:53,js : No unprocessed trips. yay!
227,1533064962.976,2018-07-31 12:22:42.976000-07:53,js : tripList.length = 0unprocessedTripList.length = 0
```

**Comment by manuelsantanap** [comment]
@shankari, I checked again when I arrived home and the app registered trips from 7:40 pm. I just sent you via email the 07/31 log and JSON files

**Comment by manuelsantanap** [comment]
@shankari , my phone was turned OFF at 10:00 am and then turned ON at 10:10 am today. When turned on the app logged me off. I clicked on Megumi's link to log in again but it didn't work.

**Comment by shankari** [comment]
wrt https://github.com/e-mission/e-mission-base/issues/3#issuecomment-409639366, I responded via email

**Comment by shankari** [comment]
wrt https://github.com/e-mission/e-mission-base/issues/3#issuecomment-409659792, the only time I have seen this happen before is when the base app was updated.
I've filed issue https://github.com/ionic-team/ionic-plugin-deploy/issues/137 against ionic almost a year ago to track this, I should probably follow up to see what they are doing about fixing it.

Has the underlying native part of the app been updated overnight? The version of the apk that I sent you was at 1.0.3 - can you make sure that you are at the same version...

**Comment by manuelsantanap** [comment]
@shankari, yes, I have version 1.0.3 installed

**Comment by shankari** [comment]
Ok so I tried reproducing, and clicking on the link for both the `cci` and `zephyr` channels just launches the app but doesn't actually download an update. This is a bit bizarre because I just tested it out recently while testing https://github.com/e-mission/cordova-jwt-auth/pull/28

**Comment by shankari** [comment]
Looking at the logs, I see messages like

```
08-04 00:58:31.053 25682 25790 I IONIC.DEPLOY.CHECK: Checking for updates
08-04 00:59:05.049 25682 25790 I IONIC.DEPLOY.CHECK: Checking for updates
08-04 01:00:05.137 25682 31955 I IONIC.DEPLOY.POST_CHECK_ERR: Unable to resolve host "api.ionicjs.com": No address associated with hostname
08-04 01:00:05.137 25682 31955 I IONIC.DEPLOY.PARSEUPDATE: Unable to check for updates.
08-04 01:00:25.169 25682 32020 I IONIC.DEPLOY.POST_CHECK_ERR: Unable to resolve host "api.ionicjs.com": No address associated with hostname
08-04 01:00:25.169 25682 32020 I IONIC.DEPLOY.PARSEUPDATE: Unable to check for updates.
```

**Comment by shankari** [comment]
Ah, the emulator in which I was testing had broken network connectivity.
I cleared and restarted the emulator and re-tried clicking on the `cci` channel link and it worked just fine.

@manuelsantanap I understand that you cannot send me the logs since you are stuck on the first screen. Can you:
- check your networking
- try to install a couple of other channels - e.g.
  - `cci`: https://e-mission.eecs.berkeley.edu/#/client_setup?new_client=cci&clear_usercache=true&clear_local_storage=true
  - `zephyr`: https://e-mission.eecs.berkeley.edu/#/client_setup?new_client=zephyr&clear_usercache=true&clear_local_storage=true
- if the networking is good and none of the channels work, send me the output of `adb logcat`.

**Comment by manuelsantanap** [comment]
@shankari, I installed the cci and zephyr channels. No problems with installation - I am sending both logs for you to have the record

**Comment by shankari** [comment]
@manuelsantanap now try the `megumibrt` channel again and let me know whether or not you are able to install it. Would be good to send combined logs (both success and failure)

**Comment by manuelsantanap** [comment]
Although the megumibrt channel works fine when installing it using adb, the app still loggs me out when restarting the phone. Once I turn it on I see the 'Deploy: Debug' message pasted below. I am also sharing the log via email and replicating the same steps with cci and zephyr channels.

![img megbrt](https://user-images.githubusercontent.com/14910471/43689560-0f9f8040-98c2-11e8-8cad-04fdb887580f.jpg)

**Comment by manuelsantanap** [comment]
Replicating megumibrt channel bug on CCI channel:

-channel correctly installed and user logged in
-when restarting the phone, the app logged me out and display "Deploy:Debug" message

@shankari, log file shared via email

**Comment by shankari** [comment]
Hm. The "Deploy: Debug" message is probably key. Do you still have the version of emTripLog from the app store installed on the phone? What happens if you do "Ignore"?

I can build and share a release apk instead. Alternatively, I can use the beta functionality in the play store. Let me just try the play store beta version first...

**Comment by manuelsantanap** [comment]
@shankari, I am still using "em-base-issue-3-token-prompt-fix.apk"

1. Reinstalled the app (test version 1.0.3) and used Megumi's link in my first test of the day
2. Logged in using my personal google email
3. Turned the phone off and then on. 
4. "Deploy Debug" message pops up and then click in 'IGNORE'
5. After that, I was left on the screen that says "you need a link or a QR code..."
6. Checked the app version and it's 1.0.3
7. Browsed for Megumi's link to use her channel and click on the link available. This action automatically opens the emTripLog app (v 1.0.3) but it does not move beyond the screen that says "you need a link or a QR code..." I clicked on the link twice but nothing happened.

I replicated the process described above (1-6) using the cci and zephyr channels and the results were the same. I am sharing the log files by email.

**Comment by manuelsantanap** [comment]
@shankari, I just replicated the test above and instead of clicking on IGNORE I clicked on UPDATE. The app does not logged me off and the version in the phone after all that is still 1.0.3. I am also sharing the log by email

**Comment by shankari** [comment]
@manuelsantanap that is very interesting. so if you apply the update, you get the correct channel UI; if you don't, the UI associated with the channel is somehow disabled.

**Comment by shankari** [comment]
I wonder if this is the same issue as https://github.com/e-mission/e-mission-phone/issues/443

**Comment by shankari** [comment]
After some digging around, it looks like the message is from the ionic deploy plugin.
https://github.com/ionic-team/cordova-plugin-ionic/commit/552435c56cf6cc1013b4dfe11d94f49aa4395f21#diff-ec4570900d598dd3bce931d96f7e95abR68

So it looks like the deploy plugin is automatically checking for updates, although we manually manage updates...

**Comment by shankari** [comment]
Yes, this is clearly an issue with deploy.
We use version 3.0.0 of the Ionic Deploy app
https://github.com/ionic-team/cordova-plugin-ionic/releases/tag/3.0.0

The last commit for which was 
https://github.com/ionic-team/cordova-plugin-ionic/commit/280bc1e71e57458da238c4fbc63e6f0f7696735f

which corresponds to PR https://github.com/ionic-team/cordova-plugin-ionic/pull/43

the last commit of which was
https://github.com/ionic-team/cordova-plugin-ionic/pull/43/commits/06c1adb70662fa4b9c4621f29a68131b97841122

browsing the tree at that commit shows us the message
https://github.com/ionic-team/cordova-plugin-ionic/blob/06c1adb70662fa4b9c4621f29a68131b97841122/src/android/IonicDeploy.java#L981

>  dlg.setMessage("A newer version of this app is available on this device.\nWould you like to update or stay on the bundled version?\n(This warning only appears in debug builds)");

So the native version of the app doesn't matter. Taking a look at the code to see when this message is generated...

**Comment by shankari** [comment]
So the message is generated from `showDebug`
https://github.com/ionic-team/cordova-plugin-ionic/blob/06c1adb70662fa4b9c4621f29a68131b97841122/src/android/IonicDeploy.java#L989

which is called (among other places) when the current `uuid` is `NO_DEPLOY_AVAILABLE`
https://github.com/ionic-team/cordova-plugin-ionic/blob/06c1adb70662fa4b9c4621f29a68131b97841122/src/android/IonicDeploy.java#L271

the current uuid is stored in local preferences
https://github.com/ionic-team/cordova-plugin-ionic/blob/06c1adb70662fa4b9c4621f29a68131b97841122/src/android/IonicDeploy.java#L182

So it looks like local preferences are not stored correctly sometimes, or are reset on reboot

**Comment by shankari** [comment]
on android, shared preferences are deleted when you clear data.
https://stackoverflow.com/questions/16566428/when-are-sharedpreferences-deleted

**Comment by shankari** [comment]
and have been reported to be flaky
https://stackoverflow.com/questions/6512917/shared-preferences-keep-disappearing#8229876

I'm using shared preferences to store the statemachine state, so if this is true, I may need to refactor some other code as well

**Comment by shankari** [comment]
so this is an ionic problem and there isn't really an easy fix.
However, I note that all this code is from the ionic plugin version 3.0.0 and was removed in 
https://github.com/ionic-team/cordova-plugin-ionic/commit/3187d253c5137a118320c2ce013cac9be211b4e8#diff-9bdc883a350801825edc331ddbea2986
The current HEAD has no `IonicDeploy.java`.

**Comment by shankari** [comment]
It looks like the deploy code is currently moved into javascript.
It still uses shared preferences to store the deploy settings, though.
https://github.com/ionic-team/cordova-plugin-ionic/blob/5.0.0/www/common.ts#L117
https://github.com/ionic-team/cordova-plugin-ionic/blob/5.0.0/src/android/IonicCordovaCommon.java#L194

so it is unclear that this will fix the issue.

**Comment by shankari** [comment]
If I file an issue with ionic, they will probably first ask me to upgrade to the latest version.
So let me update the plugin to version 5.0.0 and see if @manuelsantanap can reproduce.

**Comment by shankari** [comment]
Looking through the logs emailed by @manuelsantanap to see if there are any clues there.

```
$ grep -l IONIC ~/Downloads/logcat_megumi2.txt | wc -l
       0
```

```
$ grep -l IONIC ~/Downloads/logcat_cci2.txt | wc -l
       0
```

Aha!

```
$ grep IONIC ~/Downloads/logcat_meg8_6v1.txt | wc -l
      28
$ grep IONIC ~/Downloads/logcat_cci8_6v1.txt | wc -l
      14
$ grep IONIC ~/Downloads/logcat_zeph8_6v1.txt | wc -l
      14
```

**Comment by shankari** [comment]
Unfortunately, all of the logs relate to `DEPLOY.CHECK` and end with "No updates found". There is no `IONIC.DEPLOY.LOAD` mesage in any of them.

```
$ grep IONIC.DEPLOY.LOAD ~/Downloads/logcat_meg8_6v1.txt | wc -l
       0
$ grep IONIC.DEPLOY.LOAD ~/Downloads/logcat_cci8_6v1.txt | wc -l
       0
$ grep IONIC.DEPLOY.LOAD ~/Downloads/logcat_zeph8_6v1.txt | wc -l
       0
```

```

```

**Comment by shankari** [comment]
@manuelsantanap I am going to upload a new apk that is at the latest version of the ionic plugin (`3.0.0 -> 5.0.0`) to see if that fixes it. If it doesn't, I will complain to ionic and if they don't respond in time, I will probably fork their plugin and see if I can poke around at it to fix it.

**Comment by shankari** [comment]
the API changed between `4.0.2` and `5.0.0`. So updating to 4.0.2 first so that I don't have to change my code.

**Comment by shankari** [comment]
So something else that I noticed while looking at the ionic code.
Note that stackoverflow flags multi-threaded access to the preferences as a frequent cause of the preferences getting lost. I noticed that the preferences are initialized for every call to `execute`, but then for `unzip`, ` SharedPreferences prefs = getPreferences();` is called again.

If we did want to stay on version 4.0.2, I wonder if it would make sense to check the init code again to see if there is a race.

**Comment by shankari** [comment]
@manuelsantanap I am working on updating the plugin. In the meanwhile, can you re-generate logcat logs for the entire cycle? With logcat, unlike the e-mission logs, you can only start seeing the logs from the time that it is launched.

So to re-check what is going on, can you do the following?
- turn on adb logcat. you may want to redirect to a file for easier upload `$ adb logcat | tee /tm
- follow the two-step instructions (install emTripLog + click on link)
- save the adb logcat output to a file

- Turn on again
- turn on adb logcat 2>&1 | tee   
- *now* finally launch the app
- send *both* sets of logs (before and after the reboot)

I am actively working on switching plugins, and should finish with it sometime tomorrow

**Comment by shankari** [comment]
I upgraded to `cordova-ionic-plugin` version `5.0.0`. The diff is attached in case I want to reinstate it.
However, it does not appear to work properly with ionic1.

Here are the steps that I followed.
- Install the `em-base-issue-3-token-prompt-fix.apk`
- Click on the link for the zephyr channel
- Deploy is successful

```
DEBUG:Ionic Deploy: Update Success! true
...
Navigated to file:///data/user/0/edu.berkeley.eecs.embase/app_d57364f9-f6bf-4c06-8311-72a5c2a9c495/index.html
```

- Uninstall the `em-base-issue-3-token-prompt-fix.apk`
- Install apk with version 5.0
- Click on the link for the zephyr channel
- Deploy is successful
```
common.js:415 New file file:///data/user/0/edu.berkeley.eecs.embase/files/ionic_built_snapshots/d57364f9-f6bf-4c06-8311-72a5c2a9c495/lib/ionic-toast/dist/ionic-toast.bundle.min.js
...
common.js:415 New file file:///data/user/0/edu.berkeley.eecs.embase/files/ionic_built_snapshots/d57364f9-f6bf-4c06-8311-72a5c2a9c495/lib/angular-nvd3/index.js
...
common.js:281 Recreated app from manifest
DEBUG:Ionic Deploy: Update Success! true
VM87 common.js:350 fetched current base path as www
```
- But on reload, the `ionic-bundle.js` file is not loaded correctly.

```
ionic.bundle.js:67569 Uncaught SyntaxError: Unexpected end of input
```

which in turn means that angular is not available, and everything breaks.

```
ionic-toast.bundle.min.js:1 Uncaught ReferenceError: angular is not defined
    at ionic-toast.bundle.min.js:1
(anonymous) @ ionic-toast.bundle.min.js:1
ng-cordova.js:9 Uncaught ReferenceError: angular is not defined
    at ng-cordova.js:9
    at ng-cordova.js:7361
```

**Comment by shankari** [comment]
So I looked at the ionic.bundle.js file in the debugger, and it is missing a closing paranthesis.

```
67567: });
67568: 
67569: })(
```

instead of 

```
67567: });
67568: 
67569: })(
```

```
67567 });
67568
67569 })();
```

I checked the file just before the redirect and it was fine.

| Before update | Just before redirect | Broken after redirect |
| -------------- | -------------------- | --------------------- |
| <img width="1111" alt="screen shot 2018-08-09 at 10 30 36 am" src="https://user-images.githubusercontent.com/2423263/43915374-5c7a93c8-9bbf-11e8-8553-66f13f65c54b.png"> | <img width="1106" alt="screen shot 2018-08-09 at 10 46 12 am" src="https://user-images.githubusercontent.com/2423263/43916181-b591a404-9bc1-11e8-8ff2-2fca1f8c424f.png"> | <img width="1111" alt="screen shot 2018-08-09 at 10 46 44 am" src="https://user-images.githubusercontent.com/2423263/43916192-ba52d45e-9bc1-11e8-8736-99cb2de58cf6.png"> |



Looks like the deploy is messing up the file. Note also that I dumped all the `New file` lines from the console log and it looks like `ionic.bundle.js` is NOT in them. I have no idea what is going on.

```
$ grep ionic.bundle /tmp/downloaded_files | wc -l
       0
```

[downloaded_files.gz](https://github.com/e-mission/e-mission-base/files/2275223/downloaded_files.gz)

**Comment by shankari** [comment]
Just filed a support ticket against ionic. While waiting for their response, @manuelsantanap since you have a phone that can reproduce the issue, can you try the steps at https://github.com/e-mission/e-mission-base/issue_comments#issuecomment-411628405? I will see if I can come up with a workaround using version 3.0.

**Comment by shankari** [comment]
@manuelsantanap after you hit "update", do you still see the prompt, or does it stick now? If the update sticks the second time, a cheap workaround while waiting for ionic to respond would be to fork their plugin and always display the message (even in release builds). At least that way you can tell people to "update" if they see the message and they won't be inadvertently logged out.

@asiripanich since this seems to be related to https://github.com/e-mission/e-mission-phone/issues/443, what do you think about the workaround?

**Comment by shankari** [comment]
update from ionic:

> - Thanks for your patience - we just released v5.0.9 of cordova-plugin-ionic and we believe this version should solve the issue you reported.
> - Please try updating the plugin and ensure you've followed the steps for v1 applications as denoted in our Deploy setup document.

I will try updating to ionic v5.0.9.

@manuelsantanap were you able to reproduce the original issue? After updating to v5.0.9, I can verify whether the WSOD reported in https://github.com/e-mission/e-mission-base/issues/3#issuecomment-411832143 is fixed.

However, I have never seen the original issue (UI gets reset), so you will need to test whether it is fixed with the update.

**Comment by shankari** [comment]
This issue is now tracking two underlying issues:
- Tracking turns off. I proposed a fix 29 days ago (https://github.com/e-mission/e-mission-base/issues/3#issuecomment-408643636). @manuelsantanap can you confirm that the apk that I uploaded (https://github.com/e-mission/e-mission-base/issues/3#issuecomment-408680479) fixes the tracking issue?

- UI gets reset. I am fairly sure this is a dup of #443, tracking it in that issue.
